chia/_tests/core/consensus/test_mmr.py

```diff
@@ -0,0 +1,204 @@
+    heights = [0, 1, 1, 2, 2]  # Correct expected peak heights after each append
+
+    for leaf, expected_height in zip(leaves, heights):
+        mmr.append(leaf)
+        assert mmr.get_tree_height() == expected_height
+
+
+@pytest.mark.anyio
+async def test_mmr_block_inclusion_by_header_hash(default_1000_blocks: list[FullBlock]) -> None:
+    mmr = MerkleMountainRange()
+    header_hashes = [block.header_hash for block in default_1000_blocks]
+    for hh in header_hashes:
+        mmr.append(hh)
+    # Test first, middle, last
+    indices = [0, len(header_hashes) // 2, len(header_hashes) - 1]
+    for idx in indices:
+        hh = header_hashes[idx]
+        proof = mmr.get_inclusion_proof_by_index(idx)
+        assert proof is not None
+        peak_index, proof_bytes, other_peaks, peak = proof
+        root = mmr.get_root()
+        assert root is not None
+        assert verify_mmr_inclusion(root, hh, peak_index, proof_bytes, other_peaks, peak)
+
+
+@pytest.mark.anyio
+async def test_mmr_benchmark_default_10000_blocks(default_10000_blocks: list[FullBlock]) -> None:
+    mmr = MerkleMountainRange()
+    header_hashes = [block.header_hash for block in default_10000_blocks]
+    t0 = time.time()
+    for idx, hh in enumerate(header_hashes):
+        mmr.append(hh)
+
+    t1 = time.time()
+    append_time = t1 - t0
+    sample_indices = random.sample(range(len(header_hashes)), 1000)
+    proof_times = []
+    for idx in sample_indices:
+        hh = header_hashes[idx]
+        t_start = time.time()
+        proof = mmr.get_inclusion_proof_by_index(idx)
+        t_end = time.time()
+        proof_times.append(t_end - t_start)
+        assert proof is not None
+        peak_index, proof_bytes, other_peaks, peak = proof
+        root = mmr.get_root()
+        assert root is not None
+        assert verify_mmr_inclusion(root, hh, peak_index, proof_bytes, other_peaks, peak)
+
+    avg_proof_time = sum(proof_times) / len(proof_times)
+    logger.info(f"MMR append time for 10,000 blocks: {append_time:.4f} seconds")
+    logger.info(f"Average proof time for 1,000 random blocks: {avg_proof_time:.6f} seconds")
+
+
+# New tests for flat MMR structure
+def test_flat_mmr_basic() -> None:
+    """Test basic flat MMR operations"""
+    mmr = MerkleMountainRange()
+    assert mmr.size == 0
+    assert len(mmr.nodes) == 0
+    assert mmr.get_root() is None
+
+    # Add first leaf
+    leaf1 = random_bytes32()
+    mmr.append(leaf1)
+    assert mmr.size == 1
+    assert len(mmr.nodes) == 1  # Just the leaf
+    assert mmr.get_root() == leaf1  # Single leaf MMR root is the leaf itself
+
+    # Add second leaf
+    leaf2 = random_bytes32()
+    mmr.append(leaf2)
+    assert mmr.size == 2
+    assert len(mmr.nodes) == 3  # 2 leaves + 1 parent
+    # Root should be hash(leaf1 || leaf2)
+    from chia.util.hash import std_hash
+
+    expected_root = std_hash(leaf1 + leaf2)
+    assert mmr.get_root() == expected_root
+
+
+def test_flat_mmr_peak_positions() -> None:
```

Comment:
I think it would really help this test to have an ASCII-art illustration of the MMR.

Something like this one:
```
Height

3             14
            /    \
           /      \
          /        \
         /          \
2       6            13
       /            /   \
1     2            9     12     17
     / \          /     /  \   /
0   0   1        7     10  11 15     18
```

I also think it would be helpful to remind the reader of what the indices refer to. The input index is the *leaf*, not MMR index, right?. The return values are indices if the *peaks*, so not given numbers on the illustration above.